Modules/Devices/test/Devices.Application.Tests/Tests/Tiers/Commands/DeleteTier/HandlerTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.IntegrationEvents.Outgoing;
+using Backbone.Modules.Devices.Application.Tests.Extensions;
+using Backbone.Modules.Devices.Application.Tiers.Commands.DeleteTier;
+using Backbone.Modules.Devices.Domain.Aggregates.Tier;
+using Backbone.Modules.Devices.Domain.Entities;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+using ApplicationException = Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions.ApplicationException;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Tiers.Commands.DeleteTier;
+public class HandlerTests
+{
+    private readonly ITiersRepository _tiersRepository;
+    private readonly IEventBus _eventBus;
+    private readonly Handler _handler;
+
+    public HandlerTests()
+    {
+        _tiersRepository = A.Fake<ITiersRepository>();
+        _eventBus = A.Fake<IEventBus>();
+        _handler = CreateHandler();
+    }
+
```

Comment:
A test for the success case is missing

Reply:
Isn't `Publishes_TierDeletedIntegrationEvent_after_successful_deletion` just that?

Reply:
What if the method simply triggers the event but doesn't delete the Tier from the database?

Btw, in your negative test case, you should also check that the Tier was NOT deleted. Because it may be possible that the Delete method is called before the exception is thrown

Reply:
Makes sense, ty